akka-docs/rst/java/code/docs/http/javadsl/server/directives/RespondWithDirectivesExamplesTest.java

```diff
@@ -0,0 +1,131 @@
+        final HttpHeader gonzo = RawHeader.create("Funky-Muppet", "gonzo");
+        final HttpHeader akka = Origin.create(HttpOrigin.parse("http://akka.io"));
+
+        final Route route = path("foo", () ->
+                respondWithHeaders(Lists.newArrayList(gonzo, akka), () ->
+                        complete("beep")
+                )
+        );
+
+        testRoute(route).run(HttpRequest.GET("/foo"))
+                .assertHeaderExists("Funky-Muppet", "gonzo")
+                .assertHeaderExists("Origin", "http://akka.io")
+                .assertEntity("beep");
+
+        //#respondWithHeaders
+    }
+
+    @Test
+    public void testRespondWithDefaultHeaders() {
+        //#respondWithDefaultHeaders
+        //custom headers
+        final RawHeader blippy = RawHeader.create("X-Fish-Name", "Blippy");
+        final HttpHeader akka = Origin.create(HttpOrigin.parse("http://akka.io"));
+        final List<HttpHeader> defaultHeaders = Lists.newArrayList(blippy, akka);
+        final RawHeader elTonno = RawHeader.create("X-Fish-Name", "El Tonno");
+
+        // format: OFF
+        // by default always include the Blippy and Akka headers,
+        // unless a more specific X-Fish-Name is given by the inner route
+        final Route route =
+                respondWithDefaultHeaders(defaultHeaders, () ->            // blippy and akka
+                        respondWithHeader(elTonno, () ->                   // / el tonno
+                                path("el-tonno", () ->                     // | /
+                                        complete("¡Ay blippy!")            // | |- el tonno
+                                ).orElse(                                  // | |
+                                        path("los-tonnos", () ->           // | |
+                                                complete("¡Ay ay blippy!") // | |- el tonno
+                                        )                                  // | |
+                                )                                          // | |
+                        ).orElse(                                          // | x
+                                complete("Blip!")                          // |- blippy and akka
+                        )                                                  // x
+                );
+        //format: ON
+
+        testRoute(route).run(HttpRequest.GET("/"))
+                .assertHeaderExists("X-Fish-Name", "Blippy")
+                .assertHeaderExists("Origin", "http://akka.io")
+                .assertEntity("Blip!");
+
+        testRoute(route).run(HttpRequest.GET("/el-tonno"))
+                .assertHeaderExists("X-Fish-Name", "El Tonno")
+                .assertEntity("¡Ay blippy!");
+
+        testRoute(route).run(HttpRequest.GET("/los-tonnos"))
+                .assertHeaderExists("X-Fish-Name", "El Tonno")
+                .assertEntity("¡Ay ay blippy!");
+        //#respondWithDefaultHeaders
+    }
+}
```

Comment:
we're trying to move towards formatting with 2 spaces FYI, but don't have formatter for Java in place yet... I don't mind merging as is, but we'll reformat afterwards then